proto/src/determined/api/v1/trial.proto

```diff
@@ -520,6 +520,30 @@ message AllocationRendezvousInfoResponse {
   determined.trial.v1.RendezvousInfo rendezvous_info = 1;
 }
 
+// Gets some allocation.
+message GetAllocationRequest {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "allocation_id" ] }
+  };
+  // The id of the allocation.
+  string allocation_id = 1;
+}
+// Response to GetAllocationRequest
+message GetAllocationResponse {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "allocation_id", "state", "slots" ] }
+  };
+  // The id of the allocation.
+  string allocation_id = 1;
+  // The state of the allocation.
+  string state = 2;
```

Comment:
can we make state an enum?

https://github.com/determined-ai/determined/blob/928ba32f0e012ca2549a5b1c68d6d10e6e2e5e11/proto/src/determined/task/v1/task.proto#L11

I don't think this is the same as task state unfortently, so might need to make a new type for allocationState

```
\dT+ allocation_state 
                                               List of data types
 Schema |       Name       |  Internal name   | Size |  Elements   |  Owner   | Access privileges | Description 
--------+------------------+------------------+------+-------------+----------+-------------------+-------------
 public | allocation_state | allocation_state | 4    | PENDING    +| postgres |                   | 
        |                  |                  |      | ASSIGNED   +|          |                   | 
        |                  |                  |      | PULLING    +|          |                   | 
        |                  |                  |      | STARTING   +|          |                   | 
        |                  |                  |      | RUNNING    +|          |                   | 
        |                  |                  |      | WAITING    +|          |                   | 
        |                  |                  |      | TERMINATING+|          |                   | 
        |                  |                  |      | TERMINATED  |          |                   | 
(1 row)
```